File: 004)SimpleCalculator/app/src/main/java/com/example/a004simplecalculator/MainActivity.java
package com.example.a004simplecalculator;

import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    private EditText editTextNumber1, editTextNumber2;
    private TextView textViewResultText;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
        editTextNumber1 = findViewById(R.id.editTextNumber1);
        editTextNumber2 = findViewById(R.id.editTextNumber2);
        textViewResultText = findViewById(R.id.textViewResultText);
    }

    public void sum(View view){
        if (editTextNumber1.getText().toString().matches("") || editTextNumber2.getText().toString().matches("")){
            textViewResultText.setText("Enter Number");
        }else{
            int number1 =  Integer.parseInt(editTextNumber1.getText().toString());
            int number2 =  Integer.parseInt(editTextNumber2.getText().toString());
            int result = number1 + number2;
            textViewResultText.setText("Result: " + result);
        }
    }

    public void deduct(View view){
        if (editTextNumber1.getText().toString().matches("") || editTextNumber2.getText().toString().matches("")){
            textViewResultText.setText("Enter Number");
        }else{
            int number1 =  Integer.parseInt(editTextNumber1.getText().toString());
            int number2 =  Integer.parseInt(editTextNumber2.getText().toString());
            int result = number1 - number2;
            textViewResultText.setText("Result: " + result);
        }
    }

    public void multiply(View view){
        if (editTextNumber1.getText().toString().matches("") || editTextNumber2.getText().toString().matches("")){
            textViewResultText.setText("Enter Number");
        }else{
            int number1 =  Integer.parseInt(editTextNumber1.getText().toString());
            int number2 =  Integer.parseInt(editTextNumber2.getText().toString());
            int result = number1 * number2;
            textViewResultText.setText("Result: " + result);
        }
    }

    public void divide(View view){
        if (editTextNumber1.getText().toString().matches("") || editTextNumber2.getText().toString().matches("")){
            textViewResultText.setText("Enter Number");
        }else{
            int number1 =  Integer.parseInt(editTextNumber1.getText().toString());
            int number2 =  Integer.parseInt(editTextNumber2.getText().toString());
            int result = number1 / number2;
            textViewResultText.setText("Result: " + result);
        }
    }
}

File: 047)CalculatorKotlin/app/src/main/java/com/elyesasimsek/calculatorkotlin/MainActivity.kt
package com.elyesasimsek.calculatorkotlin

import android.os.Bundle
import android.view.View
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.databinding.DataBindingUtil
import com.elyesasimsek.calculatorkotlin.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding
    var number1: Double? = null
    var number2: Double? = null
    var result: Double? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
    }

    fun mySum(view: View){
        number1 = binding.editTextNumber1.text.toString().toDoubleOrNull()
        number2 = binding.editTextNumber2.text.toString().toDoubleOrNull()

        if (number1 != null && number2 != null){
            val result = number1!! + number2!!
            binding.textViewResult.text = "Result: ${result}"
        }else{
            binding.textViewResult.text = "Enter Number"
        }
    }

    fun mySub(view: View){
        number1 = binding.editTextNumber1.text.toString().toDoubleOrNull()
        number2 = binding.editTextNumber2.text.toString().toDoubleOrNull()

        if (number1 != null && number2 != null){
            result = number1!! - number2!!
            binding.textViewResult.text = "Result: ${result}"
        }else{
            binding.textViewResult.text = "Enter Number"
        }
    }

    fun myMul(view: View){
        number1 = binding.editTextNumber1.text.toString().toDoubleOrNull()
        number2 = binding.editTextNumber2.text.toString().toDoubleOrNull()

        if (number1 != null && number2 != null){
            result = number1!! * number2!!
            binding.textViewResult.text = "Result: ${result}"
        }else{
            binding.textViewResult.text = "Enter Number"
        }
    }

    fun myDiv(view: View){
        number1 = binding.editTextNumber1.text.toString().toDoubleOrNull()
        number2 = binding.editTextNumber2.text.toString().toDoubleOrNull()

        if (number1 != null && number2 != null){
            result = number1!! / number2!!
            binding.textViewResult.text = "Result: ${result}"
        }else{
            binding.textViewResult.text = "Enter Number"
        }
    }
}